sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -39,27 +39,36 @@ object StringUtils extends Logging {
    * throw an [[AnalysisException]].
    *
    * @param pattern the SQL pattern to convert
+   * @param escapeStr the escape string contains one character.
    * @return the equivalent Java regular expression of the pattern
    */
-  def escapeLikeRegex(pattern: String): String = {
+  def escapeLikeRegex(pattern: String, escapeStr: String): String = {
+    val escapeChar = escapeStr.charAt(0)
     val in = pattern.toIterator
     val out = new StringBuilder()
 
     def fail(message: String) = throw new AnalysisException(
       s"the pattern '$pattern' is invalid, $message")
 
     while (in.hasNext) {
-      in.next match {
-        case '\\' if in.hasNext =>
+      val cur = in.next
+      if (cur == escapeChar) {
```

Comment:
Any reason to rewrite this part from pattern-matching to if-then?

Reply:
Because only constant could after case, not variable. This will lead a compile error, when the code as follows:
```
in.next match {
  case escapeChar if in.hasNext =>
```

Reply:
You can write like this?
```
      in.next match {
        case c if c == escapeChar && in.hasNext =>
```
but, I think this rewriting has no harm. So, I leave it to other reviewers.

Reply:
OK. I think your suggestion is better. I learned it and to have a try!